src/parser/ast/ReturnStatementNode.h

```diff
@@ -39,9 +39,16 @@ class ReturnStatementNode : public StatementNode {
         if (context->m_breakpointContext->m_breakpointLocations.size() == 0) {
             ASSERT(context->m_breakpointContext->m_lastBreakpointLineOffset == 0);
             ASSERT(context->m_breakpointContext->m_breakpointLocations.size() == 0);
-            ExtendedNodeLOC sourceElementStart = context->m_codeBlock->functionStart();
-            size_t lastLineOffset = context->calculateBreakpointLineOffset(m_loc.index, sourceElementStart);
-            context->insertBreakpointAt(lastLineOffset + sourceElementStart.line, this);
+
+            InterpretedCodeBlock* interpretedCodeBlock = context->m_codeBlock;
+
+            if (interpretedCodeBlock->hasRareData() && interpretedCodeBlock->rareData()->m_debuggerLineStart != SIZE_MAX) {
+                context->insertBreakpointAt(interpretedCodeBlock->rareData()->m_debuggerLineStart, this);
```

Comment:
This if branch is for the specific case, which is a return statement of class field initializer (virtual arrow function).
So, how about adding an assert statement like `ASSERT(interpretedCodeBlock->isOneExpressionOnlyVirtualArrowFunctionExpression());` here to guard it?